### PR TITLE
Fix reset button in Try It Out form

### LINKED ISSUE #9158: "Reset" button creates invalid inputs in the Try It Out form
### Q&A (please complete the following information)
 - OS: Windows 10
 - Browser: Chrome / Firefox
 - Version: 115.0.5790.171 (Chrome)
 - Method of installation: Docker
 - Swagger-UI version: 3.18.3
 - Swagger/OpenAPI version: 3.0.3

### Content & configuration

Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.3
info:
  title: Test API
  version: 1.0.0
paths:
  /users:
    post:
      summary: Create a user
      description: Create a user, one of various ways
      requestBody:
        content:
          application/x-www-form-urlencoded:
            schema:
              $ref: '#/components/schemas/UserSource'
      responses:
        '204':
          description: Successfully opened document
        '400':
          description: Invalid request
          content:
            application/json:
              schema:
                properties:
                  output:
                    type: string
                    example: "Invalid request"
components:
  schemas:
    UserSource:
      type: object
      properties:
        name:
          description: Full name
          type: string
          example: "John Smith"
        badgeid:
          description: Badge number
          type: integer
          format: uint32
          example: 959310
        email:
          description: E-mail
          type: string
          example: "jsmith@business.com"
      minProperties: 1
      maxProperties: 1
```

### Describe the bug you're encountering
When the Reset button is used on a Try It Out form with optional parameters with default values, it creates a strange invalid state on the form.
* 'Send empty value' checkbox becomes checked, even though the form values were reset to their defaults
  -> normally, the checkbox is greyed out / off, when there is a value in the field (as it has no purpose if the field is populated)
* The request it builds does not reflect the state of the fields - it doesn't send the repopulated default values, and it also doesn't send empty values (contrary to the checkbox states). If you manually re-enter one of the form values, it will send _only_ that parameter, although the form looks like it would send all of them.

### To reproduce...

1. Try the provided yaml, e.g. in Swagger Editor online
2. Expand the single POST request
3. Click 'Try It Out'
4. Click 'Reset'

### Expected behavior
(1) Parameters should be reset to their default values in the UI
(2) "Send empty value" should be unchecked and disabled for all fields with default values set
(3) Clicking Execute should build a request with all parameters set to their values seen in the form

(Instead, only (1) is true.)

### Screenshots
bug

### PR DESCRIPTION
**Pull request by aviral-badola9**
### Description

The reset button needed the following two corrections:
1. The `oas3Actions.setRequestBodyValue` expects a orderedMap/Map and was currently being passed a string. So a conversion from string to orderedMap was added.
2. The `UPDATE_REQUEST_BODY_VALUE` action currently did not copy over the values if the value was a map assuming the user entered values will be string but that is not the case in reset, as the values will already have been converted to map. Therefore even the map values have been copied to correct key to reflect changes.
3. Furthermore, the loop to copy over the values overwrote the previous iteration changes due to it using the same unchanged variable every loop. The variable is now updated and retained every iteration to preserve changes from previous iterations.

### Motivation and Context

Fixes #9158 

### How Has This Been Tested?

I manually tested this by opening the try it out form and using the yaml file provided in the issue description.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mathis-m** [comment]
LGTM thanks for your efforts and for advancing further the reset functionality.
Didn't noticed that when implementing #6837

Maybe you could add a test for regression

**Comment by aviral-badola9** [comment]
> LGTM thanks for your efforts and for advancing further the reset functionality. Didn't noticed that when implementing #6837
> 
> Maybe you could add a test for regression

Sure, I can add that. Could you point me in the right direction of what test I can write here. I see a test for this `oas3-user-edit-request-body-flows.cy.js`. Should that be edited or a new one be added.

**Review by sabbir5564 — APPROVED** [review]

**Comment by aviral-badola9** [comment]
> mathis-m

I have added a cypress test for regression. Kindly review the changes.

File: src/core/containers/OperationContainer.jsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export default class OperationContainer extends PureComponent {
 
   onResetClick = (pathMethod) => {
     const defaultRequestBodyValue = this.props.oas3Selectors.selectDefaultRequestBodyValue(...pathMethod)
-    this.props.oas3Actions.setRequestBodyValue({ value: defaultRequestBodyValue, pathMethod })
+    this.props.oas3Actions.setRequestBodyValue({ value: OrderedMap(JSON.parse(defaultRequestBodyValue)), pathMethod })
```
Comment (glowcloud) [inline src/core/containers/OperationContainer.jsx:127]: We can change this to 
```
this.props.oas3Actions.setRequestBodyValue({ value: fromJS(JSON.parse(defaultRequestBodyValue)), pathMethod })
```
so there's no need for the `OrderedMap` import.

**Review by glowcloud — COMMENTED** [review]
Thanks for this change. 

I checked it out and it looks fine but I found more issues when using the specification from the issue itself. 
It seems like it’s because the `UserSource` schema there has `maxProperties` set to 1 / lower count than each property of the schema. Despite this, each of the example values is filled and being sent. The reset button also stays visible even without making any changes. When resetting, only the first value is reset. It seems that not resetting other values is because we check if we already added the max amount of properties in `sampleFromSchemaGeneric`.

**Comment by aviral-badola9** [comment]
> Thanks for this change.
> 
> I checked it out and it looks fine but I found more issues when using the specification from the issue itself. It seems like it’s because the `UserSource` schema there has `maxProperties` set to 1 / lower count than each property of the schema. Despite this, each of the example values is filled and being sent. The reset button also stays visible even without making any changes. When resetting, only the first value is reset. It seems that not resetting other values is because we check if we already added the max amount of properties in `sampleFromSchemaGeneric`.

Thanks for the review @glowcloud . I will take a look at this issue.

**Comment by glowcloud** [comment]
I created a separate issue for `maxProperties` not being validated by Swagger UI https://github.com/swagger-api/swagger-ui/issues/9675

The issue with only resetting some parameters seems related to this PR/issue, so leaving it here for now.

**Comment by aviral-badola9** [comment]
> I created a separate issue for `maxProperties` not being validated by Swagger UI #9675
> 
> The issue with only resetting some parameters seems related to this PR/issue, so leaving it here for now.

Have used the respectXML prop to bypass maxProperties check. The reset button now only appears if the body is actually edited and all fields are reset. Please review the changes.  @glowcloud

**Pull request closed without merging**